wremnants/theory_tools.py

```diff
@@ -190,25 +200,23 @@ def define_pdf_columns(df, dataset, pdfs, noAltUnc):
         entries = 1 if i != 0 and noAltUnc else pdfInfo["entries"]
         start = 0 if "first_entry" not in pdfInfo else pdfInfo["first_entry"]
 
-        df = df.Define(tensorName, f"auto res = wrem::clip_tensor(wrem::vec_to_tensor_t<double, {entries}>({pdfBranch}, {start}), 10.); res = nominal_weight/nominal_pdf_cen*res; return res;")
+        if i == 0:
+            df = df.Define(tensorName, f"auto res = wrem::clip_tensor(wrem::vec_to_tensor_t<double, {entries}>({pdfBranch}, {start}), 10.); res = nominal_weight*res; return res;")
+            df = df.Define("nominal_pdf_cen", f"{tensorName}(0)/nominal_weight")
```

Comment:
doesn't the tensor already contain the ratio to nominal? (ie is it really supposed to be divided by nominal_weight here?)

Reply:
I think this is to get the pure central pdf weight to use it to rederive the original nominal weight for the other pdf sets. 
Doing the redefine after the for loop (applying the nominal pdf weight later) would make code more easy

Reply:
I propose:
```
        df = df.Define(tensorName, f"auto res = wrem::clip_tensor(wrem::vec_to_tensor_t<double, {entries}>({pdfBranch}, {start}), 10.); res = nominal_weight*res; return res;")

        if i == 0:
            tensorNameNominal = tensorName

        df = df.Define(tensorASName, "Eigen::TensorFixedSize<double, Eigen::Sizes<2>> res; "
                f"res(0) = nominal_weight*{pdfInfo['alphas'][0]}; "
                f"res(1) = nominal_weight*{pdfInfo['alphas'][1]}; "
                "return wrem::clip_tensor(res, 10.)")

    # now set the nominal pdf
    df = df.Redefine("nominal_weight", f"{tensorNameNominal}(0)")

    return df
```

Reply:
You implemented this, right?

Reply:
Yes